doc/specs/#888 - Com Api.md

```diff
@@ -0,0 +1,812 @@
+        PackageMatchFilter MatchCriteria { get; };
+    }
+    /// IMPLEMENTATION NOTE: SearchResult from AppInstallerRepositorySearch.h
+    /// Search result data returned from FindPackages
+    runtimeclass FindPackagesResult
+    {
+        FindPackagesResult();
+
+        /// The full set of results from the search.
+        Windows.Foundation.Collections.IVectorView<ResultMatch> Matches { get; };
+
+        /// If true, the results were truncated by the given SearchRequest::MaximumResults.
+        /// USAGE NOTE: Windows Package Manager does not support result pagination, there is no way to continue getting more results.
+        Boolean IsTruncated{ get; };
+    }
+
+    /// Options for FindPackages
+    runtimeclass FindPackagesOptions
+    {
+        FindPackagesOptions();
+
+        /// DESIGN NOTE: 
+        /// This class maps to SearchRequest from  AppInstallerRepositorySearch.h 
+        /// That class is a container for data used to filter the available manifests in an app catalog.
+        /// Its properties can be thought of as:
+        /// (Query || Inclusions...) && Filters...
+        /// If Query and Inclusions are both empty, the starting data set will be the entire database.
+        /// Everything && Filters...
+        /// That has been translated in this api so that 
+        /// Query is PackageMatchField::AppCatalogDefined and PackageMatchFilter.IsAdditive = true
+        /// Inclusions are PackageMatchFilter.IsAdditive = true
+        /// Filters are PackageMatchFilter.IsAdditive = false
+
+        /// Filters to find packages
+        /// USAGE NOTE: Only one filter with PackageMatchField::AppCatalogDefined is allowed.
+        Windows.Foundation.Collections.IVector<PackageMatchFilter> Filters;
+
+        /// Restricts the length of the returned results to the specified count.
+        UInt32 Limit;
+
+        /// Overrides the default search behavior for this search if the catalog is a composite catalog.
+        /// IMPLEMENTATION NOTE: Windows Package Manager currently only supports setting the search behavior when creating the source
+        /// so this is new work but is expected to be very small.
+        CompositeSearchBehavior CompositeSearchBehavior;
+    }
+
+    /// IMPLEMENTATION NOTE: ISource from AppInstallerRepositorySource.h
+    /// A catalog for searching for packages
+    runtimeclass AppCatalog
+    {
+        AppCatalog();
+
+        /// Gets a value indicating whether this app catalog is a composite of other app catalogs,
+        /// and thus the packages may come from disparate app catalogs as well.
+        Boolean IsComposite { get; };
+        /// The details of the app catalog if it is not a composite.
+        AppCatalogDetails Details { get; };
+
+        /// Opens a catalog. Required before searching. For remote catalogs (i.e. note Installed and Installing) this may require downloading information from a server.

```

Comment:
This should probably return an `AppCatalogOpenResult` that tells you what happened (like network disconnect, access denied, invalid format, success, etc.)

Reply:
Can do - I've skipped this for now though pending the question about whether IAsyncAction error code is sufficient.